### PR TITLE
add_process_metadata processor adds container id even if process metadata not accessible

### PR DESCRIPTION
**Pull request by jtinkus**
- Enhancement
## What does this PR do?

Changed add_process_metadata processor to get both process metadata and container id and not error out if only one is available. 

## Why is it important?

If container is non-privileged, then process metadata for external processes is not fully readable (no access to /proc/pid/exe and /proc/pid/cwd) and code errors out before even trying to get container id. Same time container id is still accessible in /proc/pid/cgroup file. Now process metadata is skipped for such processes, but container id is still added. 

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [] I have made corresponding changes to the documentation
- [] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

Minor fix, probably no need for doc or changelog.
## Author's Checklist

- [ ]

## Related issues

-

### CODE REVIEW

**Comment by jtinkus** [comment]
@exekias proposal for minor fix in the code you have reviewed before.

File: libbeat/processors/add_process_metadata/add_process_metadata.go [hunk 1]
```diff
@@ -190,15 +190,29 @@ func (p *addProcessMetadata) enrich(event common.MapStr, pidField string) (resul
...
 		p.log.Debugf("failed to get process metadata for PID=%d: %v", pid, err)
-		return nil, ErrNoProcess
+		meta = common.MapStr{}
+	} else {
+		meta = metaPtr.fields
 	}
-	meta := metaPtr.fields
 
-	if err = p.enrichContainerID(pid, meta); err != nil {
-		return nil, err
```
Comment (exekias) [inline libbeat/processors/add_process_metadata/add_process_metadata.go:201]: the first part looks great, but why is this change needed? I think that `p.enrichContainerID` would do what you wanted?

Comment (jtinkus) [reply libbeat/processors/add_process_metadata/add_process_metadata.go:201]: a. p.enrichContainerID was function having only side effect - changing external object. Not a good functional style, hard to follow. IMHO.
b. It also set empty cid to meta, why add empty container id to event?

Actually in general I find handling container id separately from other process metadata a bit hard to follow style. Logically it is same as extracting other process metadata, and it is extracted also from same place - files under /proc/pid. So it would fit nicely into go-sysinfo package (iirc) where all other metadata is extracted from process files.

Comment (jtinkus) [reply libbeat/processors/add_process_metadata/add_process_metadata.go:201]: @exekias  should i add changelog entry to CHANGELOG.next? 
is it ok to fix also line for main feature we are modifying? this sentence does not seem to be correct in my opinion:
- Add capability of enrich `container.id` with process id in `add_process_metadata` processor {pull}15947[15947]

IMO Container metadata is enriched with container id based on process id.
:)

Comment (exekias) [reply libbeat/processors/add_process_metadata/add_process_metadata.go:201]: Ok the change LGTM then. About changelog:

This feature was already released so updating the changelog will have no effect. In the same sense, as this is changing the current behaviour we will need a new line :)

Comment (jtinkus) [reply libbeat/processors/add_process_metadata/add_process_metadata.go:201]: stuff in changelog.next is released? what file i need to add my line then?

Comment (jtinkus) [reply libbeat/processors/add_process_metadata/add_process_metadata.go:201]: @exekias updated changelog as stated in dev guide.

**Comment by ChrsMark** [comment]
jenkins run the tests please

**Comment by jtinkus** [comment]
Seems that PR is now waiting approval from @exekias

**Comment by ChrsMark** [comment]
@exekias will be out for some days so I'm merging this one to avoid waiting.
Thank you @jtinkus for contributing this! Could you also share your manual testing process on the description of this PR (`How to test this PR locally`)?
